### PR TITLE
Check Azure credential when creating new Azure Cache object

### PR DESCRIPTION
**Pull request by renxinhe**
# Summary

Previously, creating a new Azure Cache object (i.e. `containerURL`) doesn't check the Azure credential until a web request is sent to our TLS server. In other words, credential is *lazily* checked. This means that if we enter the wrong Azure Blob Storage credential, the TLS server would crash when the first web request come in, instead of when the server boot up (See issue #1).

In this PR, we additionally call Azure API's `GetAccountInfo` to trigger the authentication. In fact, we don't care about the response; we only care if it returns an error.

# Testing

Added new unit test `TestNew_BadCredential`. Ran all unit tests with my own Azure Blob Storage account. All tests pass:

image

Also tested this change in our server code. `azcertcache.New` successfully returned the correct authentication error, and we were able to fallback to local cache.

```
$ make run
2022/06/07 06:45:47 Checking Azure credential
2022/06/07 06:45:47 Error connecting to TLS certificate cache in Azure Blob Storage: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /home/jim/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.15.0/azblob/zc_storage_error.go:42
===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====
Description=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:7763873d-501e-0032-053a-7a4066000000
Time:2022-06-07T06:45:47.7099181Z, Details: 
   AuthenticationErrorDetail: The MAC signature found in the HTTP request '4XGdIYVNPtQGoKC9Al3sBcwSv3fQxR3BvLmeo7+ZUH8=' is not the same as any computed signature. Server used following string to sign: 'GET











x-ms-client-request-id:5ac543f3-b592-40d7-4629-cd95b71d5882
x-ms-date:Tue, 07 Jun 2022 06:45:47 GMT
x-ms-version:2020-10-02
<REDACTED>
comp:properties
restype:account
timeout:61'.
   Code: AuthenticationFailed
   GET https://<REDACTED>.blob.core.windows.net/<REDACTED>comp=properties&restype=account&timeout=61
   Authorization: REDACTED
   User-Agent: [Azure-Storage/0.15 (go1.18.1; linux)]
   X-Ms-Client-Request-Id: [5ac543f3-b592-40d7-4629-cd95b71d5882]
   X-Ms-Date: [Tue, 07 Jun 2022 06:45:47 GMT]
   X-Ms-Version: [2020-10-02]
   --------------------------------------------------------------------------------
   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
   Content-Length: [775]
   Content-Type: [application/xml]
   Date: [Tue, 07 Jun 2022 06:45:47 GMT]
   Server: [Microsoft-HTTPAPI/2.0]
   X-Ms-Error-Code: [AuthenticationFailed]
   X-Ms-Request-Id: [7763873d-501e-0032-053a-7a4066000000]


 
Falling back to local disk certificate cache
```

# List of changes

## Library
- Call `GetAccountInfo` after creating a `containerURL` object
- Version bump `github.com/Azure/azure-storage-blob-go` from v0.0.0 (2019) to v0.15.0 (2022)
- Using `azblob`'s minimum go requirement (i.e. go 1.15) in our go.mod file as well

## Unit tests
- Added new unit test `TestNew_BadCredential`
- Using `testify` library for asserts; previous logic was buggy when it comes to interface nil check due to lack of reflection and generic casting
- Lifted out account name and key into const variables
- Passing in account key into `newCache` so we can test bad credential use case
- Each unit test now creates a separate container with *unique names*; now we can quickly run all the tests in parallel

### CODE REVIEW

File: azcertcache_test.go [hunk 1]
```diff
@@ -2,146 +2,156 @@ package azcertcache_test
...
+// Bad account key encoded in base64 that would fail the authentication check
+const badAccountKey = "YmFkY3JlZGVudGlhbA==" // base64: "badcredential"
 
-func expectEquals(t *testing.T, v interface{}, expected interface{}) {
-	if !reflect.DeepEqual(v, expected) {
-		t.Errorf("should be %v, but was %v", expected, v)
-	}
-}
+func newCache(t *testing.T, containerPrefix string, accountKey string) (*azcertcache.Cache, error) {
+	// Append test name after the container prefix to make them different.
```
Comment (christodenny) [inline azcertcache_test.go:25]: Can this still have namespace collisions if multiple people run the same test at the same time? Maybe also add a random string to the container name?

File: azcertcache_test.go [hunk 2]
```diff
@@ -2,146 +2,156 @@ package azcertcache_test
...
+		assert.Nil(t, err)
+	})
+	return cache, nil
 }
 
 func TestNew(t *testing.T) {
-	cache := newCache(t, "testcontainer")
-	expectNotNil(t, cache)
+	cache, _ := newCache(t, "testcontainer", accountKey)
+	assert.NotNil(t, cache)
```
Comment (christodenny) [inline azcertcache_test.go:53]: nit, also assert.Nil(err) for completeness?

File: azcertcache_test.go (hunk 1 above)
Comment (renxinhe) [reply azcertcache_test.go:25]: Good point, but this is outside of expected use case. Doubt we are going to have people running tests at the same time. Besides, I don't even know if the storage account in this test is still running. 

I'll add comments to tell people to use their own Azure storage account name and keys at the top of this file.

Comment (goenning) [reply azcertcache_test.go:25]: that is not a real azure account, the tests target a local container (described on docker-compose.yml)

**Comment by goenning** [comment]
this repo could really use GitHub actions for automated PR tests in case someone is interested on :)

**Pull request merged**